qiskit/circuit/library/evolved_operator_ansatz.py

```diff
@@ -152,6 +165,32 @@ def preferred_init_points(self):
             self._build()
             return np.zeros(self.reps * len(self.operators), dtype=float)
 
+    def _evolve_operator(self, operator, time):
+        from qiskit.opflow import OperatorBase, EvolutionBase
+        from qiskit.extensions import HamiltonianGate
+
+        if isinstance(operator, OperatorBase):
+            if not isinstance(self.evolution, EvolutionBase):
+                raise QiskitError(
+                    "If qiskit.opflow operators are evolved the evolution must be a "
+                    f"qiskit.opflow.EvolutionBase, not a {type(self.evolution)}."
+                )
```

Comment:
What does this mean for UCC? That generated PauliSumOps and Nature is setting migration over to SparsePauliOp. But until the former goes away we can still use opflow internally here until its removed if there is no reasonable way to support these otherwise.

Reply:
Since UCC is only using first order Trotter expansion, we could just remove the explicit `evolution=PauliTrotterEvolution()` there and wouldn't trigger any deprecation. Otherwise, the code would have to updated to pass an `EvolutionSynthesis` if a `SparsePauliOp` is used as operator.